app/full_text_search.py

```diff
@@ -166,31 +210,37 @@ def create_document(record_id, repo, **kwargs):
     fields = []
 
     # Add repo and record_id to fields
+    repo = person.repo
+    record_id = person.record_id
     doc_id = repo + ':' + record_id
     fields.append(appengine_search.TextField(name='repo', value=repo))
     fields.append(appengine_search.TextField(name='record_id', value=record_id))
 
-    # Add name and location romanized by unidecode
-    for field in kwargs:
-        romanized_value = script_variant.romanize_word(kwargs[field])
-        fields.append(
-            appengine_search.TextField(name=field, value=romanized_value))
-
-    # Add name romanized by japanese name dictionary
-    fields.extend(create_jp_name_fields(
-        given_name=kwargs['given_name'],
-        family_name=kwargs['family_name'],
-        full_name=kwargs['full_name'],
-        alternate_names=kwargs['alternate_names']))
-
-    # Add location romanized by japanese location dictionary
-    fields.extend(create_jp_location_fields(
-        home_street=kwargs['home_street'],
-        home_city=kwargs['home_city'],
-        home_state=kwargs['home_state'],
-        home_postal_code=kwargs['home_postal_code'],
-        home_neighborhood=kwargs['home_neighborhood'],
-        home_country=kwargs['home_country']))
+    # romanize name method by unidecode or japanese name dictionary.
```

Comment:
Please explain why, instead of what you are doing, which is obvious from code in this case.